### PR TITLE
testing: fix pexpect hang

### LINKED ISSUE #7913: testing: test_pdb_unittest_postmortem get stuck with pexpect
This causes the ubuntu-py37 CI job to get stuck and timeout after 6 hours. This job runs `tox -e py37-lsof-numpy-pexpect` but I can reproduce locally with `tox -e py38-pexpect` so it's not about the python version or lsof/numpy.

When running just `tox -e py-pexpect -- testing/test_debugging.py` it does not happen; so it's only with some interaction with other tests.

The pexpect timeout evidentially doesn't kick in.

Don't have a useful traceback.

Bisected to 69419cb70007514e5061f3b8d359b7da9dff82dd (new pytester fixture) - cc @nicoddemus.

That's what I had time to investigate for now.

### PR DESCRIPTION
**Pull request by bluetech**
This fixes #7913 (hopefully). There are some other commits I made to testing while debugging this but they are not critical.

### CODE REVIEW

File: src/_pytest/pytester.py [hunk 1]
```diff
@@ -707,8 +707,11 @@ def __take_sys_modules_snapshot(self) -> SysModulesSnapshot:
         # Some zope modules used by twisted-related tests keep internal state
         # and can't be deleted; we had some trouble in the past with
         # `zope.interface` for example.
+        #
+        # Preserve readline due to https://bugs.python.org/issue41033.
+        # pexpect issues a SIGWINCH.
         def preserve_module(name):
-            return name.startswith("zope")
+            return name.startswith("zope") or name.startswith("readline")
```
Comment (RonnyPfannschmidt) [inline src/_pytest/pytester.py:714]: Starts supports a tuple of variants

**Review by RonnyPfannschmidt — APPROVED** [review]
Lovely

File: src/_pytest/pytester.py (hunk 1 above)
Comment (bluetech) [reply src/_pytest/pytester.py]: Updated, thanks.

**Comment by bluetech** [comment]
I'll merge already to fix the broken CI, but let me know if there are any further comments.

**Comment by nicoddemus** [comment]
Awesome, thanks @bluetech!

**Pull request merged**